internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -755,6 +776,20 @@ func (r *BMACReconciler) reconcileSpokeBMH(ctx context.Context, log logrus.Field
 		Namespace: agent.Spec.ClusterDeploymentName.Namespace,
 		Name:      fmt.Sprintf(adminKubeConfigStringTemplate, cd.Name),
 	}
+	isNonePlatform, requeue, err := r.isNonePlatform(ctx, cd)
+	if err != nil {
+		log.WithError(err).Error("Failed to check if platform is none")
+		if requeue {
+			return reconcileRequeue{requeueAfter: DEFAUAT_BMH_REQUEUE_DURATION}
```

Comment:
I'm trying to think about this a bit more and I don't think we need to requeue. I know we briefly discussed this in the previous version of the PR so, please, don't hate me :hear_no_evil: 

If the ACI doesn't exist (which it should if we consider this is day2), I think it's fine for us to just error out. Using requeue is not recommended unless necessary. 

I think it's fine to log the error if ACI doesn't exist and return `reconcileComplete` This is because we have a watch on `ClusterDeployment` which would trigger a new loop if something changes on that front.

Reply:
If ACI doesn't exist requeue is false so we do not requeue.  Requeue is done if there is an error on Get

Reply:
> If ACI doesn't exist requeue is false so we do not requeue. Requeue is done if there is an error on Get

This is exactly what I mean. The get could also fail because the ACI was specified in the CD but it doesn't exist.

So, if we fail to get ACI for whatever reason, I think we should log the error but skip the requeue. We have the watches in place which should trigger a reconcile if needed.

Reply:
OK

Reply:
So what is wrong now ?